File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/FViews/FTextView.java
package io.github.gofaith.faithdroid.FViews;

import android.graphics.Color;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import org.json.JSONArray;
import org.json.JSONTokener;

import faithdroid.Faithdroid;
import io.github.gofaith.faithdroid.UI.AttrGettable;
import io.github.gofaith.faithdroid.UI.AttrSettable;
import io.github.gofaith.faithdroid.UI.UIController;

public class FTextView extends FView implements AttrGettable,AttrSettable {
    private TextView v;

    public FTextView(UIController c) {
        parrentController=c;
        v = new TextView(parrentController.activity);
        view=v;
    }
    @Override
    public String getAttr(String attr) {
        switch (attr) {
            case "Visibility":
                return getVisibility();
                //--------------------------------------------
            case "Text":
                return v.getText().toString();
        }
        return "";
    }

    @Override
    public void setAttr(String attr, final String value) {
        switch (attr) {
            case "BackgroundColor":
                setBackgroundColor(value);
                break;
            case "Background":
                setBackground(value);
                break;
            case "Size":
                parseSize(parrentController.activity, v, value);
                break;
            case "Visibility":
                setVisibility(value);
                break;
            case "Padding":
                setPadding(value);
                break;
            case "Margin":
                setMargin(value);
                break;
            case "LayoutGravity":
                setLayoutGravity(value);
                break;
            case "Elevation":
                setElevation(value);
                break;
                // ----------------------------------------------------------
            case "Text":
                if (value!=null)
                    v.setText(value);
                break;
            case "TextColor":
                try {
                    v.setTextColor(Color.parseColor(value));
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "TextSize":
                try {
                    v.setTextSize(dp2pixel(parrentController.activity,Float.valueOf(value)));
                } catch (Exception e) {
                    e.printStackTrace();
                    return;
                }
                break;
            case "OnClick":
                v.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        Faithdroid.triggerEventHandler(value,"");
                    }
                });
                break;
        }
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/FViews/FTextView.kt
package io.github.stevenzack.faithdroidapp.FViews

import android.graphics.Color
import android.view.View
import android.widget.TextView
import faithdroid.Faithdroid
import io.github.stevenzack.faithdroidapp.UI.AttrGettable
import io.github.stevenzack.faithdroidapp.UI.AttrSettable
import io.github.stevenzack.faithdroidapp.UI.FView
import io.github.stevenzack.faithdroidapp.UI.Toolkit.dp2pixel
import io.github.stevenzack.faithdroidapp.UI.UIController

class FTextView(c: UIController) : FView(), AttrGettable, AttrSettable {
    private val v: TextView

    init {
        parentController = c
        v = TextView(parentController!!.activity)
        view = v
    }

   override fun getAttr(attr: String): String {
        val str = getUniversalAttr(attr)
        if (str != null) {
            return str
        }
        when (attr) {
            //--------------------------------------------
            "Text" -> return v.text.toString()
        }
        return ""
    }

   override fun setAttr(attr: String, value: String) {
        if (setUniversalAttr(attr, value)) {
            return
        }
        when (attr) {
            // ----------------------------------------------------------
            "Text" -> if (value != null)
                v.text = value
            "TextColor" -> try {
                v.setTextColor(Color.parseColor(value))
            } catch (e: Exception) {
                e.printStackTrace()
            }

            "TextSize" -> try {
                v.textSize = dp2pixel(parentController!!.activity, java.lang.Float.valueOf(value))
            } catch (e: Exception) {
                e.printStackTrace()
                return
            }

            "OnClick" -> v.setOnClickListener { Faithdroid.triggerEventHandler(value, "") }
        }
    }
}
